src/mds/MDLog.cc

```diff
@@ -827,7 +828,7 @@ void MDLog::_trim_expired_segments()
   // trim expired segments?
   bool trimmed = false;
   while (!segments.empty()) {
-    LogSegment *ls = segments.begin()->second;
+    LogSegmentRef& ls = segments.begin()->second;
```

Comment:
```suggestion
    auto ls = segments.begin()->second;
```

You don't want a reference since you plan to delete it via `segments.erase`. It's generally not good to have a reference to something that will be deleted.